File: r/roman-numerals/_iterations/1/roman-numerals.R
roman <- function(arabic) {
  result   <- c()
  arabics  <- c(1000, 900, 500, 400, 100, 90,
                50, 40, 10, 9, 5, 4, 1)
  romans   <- c("M", "CM", "D", "CD", "C", "XC",
                "L", "XL", "X", "IX", "V", "IV", "I")

  for (index in seq_along(arabics)) {
    current <- arabics[index]
    while (current <= arabic) {
      arabic <- arabic - current
      result <- append(result, romans[index])
    }
  }
  paste(result, sep = "", collapse = "")
}

File: nim/roman-numerals/_iterations/1/roman_numerals.nim
const arabicToRoman = {
  1000: "M", 900: "CM", 500: "D", 400: "CD",
   100: "C",  90: "XC",  50: "L",  40: "XL",
    10: "X",   9: "IX",   5: "V",   4: "IV",
     1: "I",
}

func roman*(decimal: int): string =
  result = ""
  var decimal = decimal
  for (arabic, roman) in arabicToRoman:
    while decimal >= arabic:
      decimal -= arabic
      result.add roman